Many ATM tweaks including ATM UI overhaul

Ravensdale opened the pull request:
- Updated ATM interface to use the pseudo-Nano HTML UI.

- ATMs now correctly count 'number of tries remaining', locking down should '1 tries remaining' fails instead of allowing one last attempt at '0 tries remaining'.

- Re-worded some of what the ATM displays, primarily those involving how to login.
ATMs will now announce what is needed to login to an account, pulling up a PIN entry field when necessary and indirectly fixing #6836 which was caused by ATMs logging into accounts by card scanning and ignoring account# and PIN fields entirely if account security was low enough. 

- Needless to say, fixes #6836

- emags will actually work on ATMs now.

- Removed a TODO list that was actually done :open_mouth: 

UI code was inspired by the solar control console

Ravensdale commented:
Well, tweaks and fixes, feh

Techhead0 commented on code/modules/economy/ATM.dm:
```diff
@@ -120,103 +110,127 @@ log transactions
 
+	if(get_dist(src,user) <= 1)
+		//make the window the user interacts with, divided out into welcome message, card 'slot', then login/data screen
+		var/t = ""
```
You should make `t` a list and use `popup.set_content(list2text(t))` below. Mostly because BYOND string concatenation sucks for long strings.

PsiOmegaDelta replied:
`` jointext(t,null)```. ```list2text()``` is deprecated.

Techhead0 replied:
My bad. Fixed.

Ravensdale commented:
Alrighty, made the super-string a list as per Techhead's suggestion

atlantiscze commented:
It'd be nice if you added ability for synthetics to interface with it in some way. Even if it was limited to traitor/malf AIs, it's open up new opportunities.

mwerezak commented on code/modules/economy/ATM.dm:
```diff
@@ -33,10 +24,12 @@ log transactions
 	var/editing_security_level = 0
 	var/view_screen = NO_SCREEN
 	var/datum/effect/effect/system/spark_spread/spark_system
+	var/obj/item/weapon/card/id/scanned_card
```
As far as I can tell this variable is only read from inside Topic(), it seems like it should be a local variable there.

Techhead0 replied:
If it needs to persist across multiple `Topic()` procs, then doesn't it need to be defined here?

mwerezak replied:
It doesn't appear to need to persist though, it's set and then immediately used. Unless I'm missing something?

Ravensdale replied:
There's also `scan_user()` that assigns the value, unless I'm to force all of that into `topic()` as a whole, this was simplest.

Techhead0 replied:
For the `scan_user()` case, you could always make `scan_user()` return `I` and then do `scanned_card = scan_user()`.

Ravensdale commented:
atlantiscze: This was honestly a fix primarily for #6836 that expanded to a UI overhaul to help facilitate the fix. I considered allowing synthetic interaction but wanted to try and keep focused on the fix first and foremost. the e-mag fix took five seconds so I went ahead and did that.

Honestly, for AI interaction I'm more tempted to bring it to a discussion on the main BS12 forums than just slip it in as I see arguments for and against.

Ravensdale commented:
Moved a var to the only place it should be needed, changed a proc to accommodate as per @mwerezak 's and @Techhead0 's  suggestions.

PsiOmegaDelta commented:
:+1:

GinjaNinja32 commented:
- Logging in on medium security without an ID requires you to input the account number twice; once alone, and again with the PIN.
- Maximum security doesn't actually require account number, just card and pin

Kasuobes commented:
Suggestion: allow ATMs to vend out coins. Right now there are a total of 4 silver coins on the station and more than enough premium items that can be bought (eg. cigar cases), With the coin mint removed from the game a new source of coin would be nice.

Recommend setting the conversion ratio to 200 thalers per silver coin for now.

Ravensdale commented:
Changed ATMs to now require an ID in the machine and an account # manually entered to login at maximum security settings.

Re-worded and changed the login page to let the user know they will need to re-enter some information if more information is required.

GinjaNinja32 commented on code/modules/economy/ATM.dm:
```diff
@@ -64,29 +56,24 @@ log transactions
 	if(istype(I, /obj/item/weapon/card))
 		if(emagged > 0)
 			//prevent inserting id into an emagged ATM
 			user << "\red \icon[src] CARD READER ERROR. This system has been compromised!"
```
Given you're making significant edits to this file, replacement of color escapes would be excellent :)

GinjaNinja32 commented on code/modules/economy/ATM.dm:
```diff
@@ -289,7 +318,7 @@ log transactions
 								playsound(src, 'sound/machines/buzz-sigh.ogg', 50, 1)
 						else
-							usr << "\red \icon[src] incorrect pin/account combination entered."
+							usr << "\red \icon[src] Unable to log in to account, additional information may be required."
```
Please replace this `\red`.

Ravensdale commented:
Changes colour escapes ( \red ) to <span>s as @GinjaNinja32 suggested.

GinjaNinja32 commented:
:+1:, though defines for security level rather than magic 0/1/2 would be nice :P
